furnace-service-provider/src/main/java/org/jboss/windup/web/furnaceserviceprovider/FurnaceExtension.java

```diff
@@ -0,0 +1,392 @@
+                }
+
+                @Override
+                public String toString()
+                {
+                    return "FurnaceInjectionTarget";
+                }
+            };
+
+            abd.addBean(new Bean<Furnace>()
+            {
+                @Override
+                public Class<?> getBeanClass()
+                {
+                    return Furnace.class;
+                }
+
+                @Override
+                public Set<InjectionPoint> getInjectionPoints()
+                {
+                    return it.getInjectionPoints();
+                }
+
+                @Override
+                public boolean isNullable()
+                {
+                    return false;
+                }
+
+                @Override
+                public Set<Type> getTypes()
+                {
+                    return Collections.singleton(Furnace.class);
+                }
+
+                @Override
+                public Set<Annotation> getQualifiers()
+                {
+                    Set<Annotation> qualifiers = new HashSet<>();
+                    qualifiers.add(new AnnotationLiteral<Default>()
+                    {
+                    });
+                    qualifiers.add(new AnnotationLiteral<Any>()
+                    {
+                    });
+                    return qualifiers;
+                }
+
+                @Override
+                public Class<? extends Annotation> getScope()
+                {
+                    return ApplicationScoped.class;
+                }
+
+                @Override
+                public String getName()
+                {
+                    return "furnace";
+                }
+
+                @Override
+                public Set<Class<? extends Annotation>> getStereotypes()
+                {
+                    return Collections.emptySet();
+                }
+
+                @Override
+                public boolean isAlternative()
+                {
+                    return false;
+                }
+
+                @Override
+                public Furnace create(CreationalContext<Furnace> creationalContext)
+                {
+                    Furnace instance = it.produce(creationalContext);
+                    it.inject(instance, creationalContext);
+                    it.postConstruct(instance);
+                    return instance;
+                }
+
+                @Override
+                public void destroy(Furnace instance, CreationalContext<Furnace> creationalContext)
+                {
+                    it.preDestroy(instance);
+                    it.dispose(instance);
+                    creationalContext.release();
+                }
+
+                @Override
+                public String toString()
+                {
+                    return "FurnaceBean";
+                }
+            });
+        }
+        catch (Throwable t)
+        {
+            t.printStackTrace();
```

Comment:
Maybe, although I am having a hard time thinking of how it wouldn't fail anyway.